File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        def digisumprod(n):
            sum = 0
            prod = 1

            while n > 0:
                digi = n % 10
                sum += digi
                prod *= digi
                n //= 10
            
            return sum+prod
        

        return n%digisumprod(n)==0

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    int digisumprod(int n) {
        if (n <= 9)
            return 2 * n;

        int sum = 0;
        int prod = 1;

        while (n > 0) {
            int digi = n % 10;
            sum += digi;
            prod *= digi;
            n /= 10;
        }

        return sum + prod;
    }

    bool checkDivisibility(int n) { return n % digisumprod(n) == 0; }
};